packages/libraries/cli/src/commands/whoami.ts

```diff
@@ -95,62 +94,44 @@ export default class WhoAmI extends Command<typeof WhoAmI> {
+        showAll: flags.all,
+      },
     });
 
-    if (result.tokenInfo.__typename === 'TokenInfo') {
-      const { tokenInfo } = result;
-      const { organization, project, target } = tokenInfo;
-
-      const organizationUrl = `https://app.graphql-hive.com/${organization.slug}`;
-      const projectUrl = `${organizationUrl}/${project.slug}`;
-      const targetUrl = `${projectUrl}/${target.slug}`;
-
-      const access = {
-        yes: Texture.colors.green('Yes'),
-        not: Texture.colors.red('No access'),
-      };
-
-      const print = createPrinter({
-        'Token name:': [Texture.colors.bold(tokenInfo.token.name)],
-        ' ': [''],
-        'Organization:': [
-          Texture.colors.bold(organization.slug),
-          Texture.colors.dim(organizationUrl),
-        ],
-        'Project:': [Texture.colors.bold(project.slug), Texture.colors.dim(projectUrl)],
-        'Target:': [Texture.colors.bold(target.slug), Texture.colors.dim(targetUrl)],
-        '  ': [''],
-        'Access to schema:publish': [tokenInfo.canPublishSchema ? access.yes : access.not],
-        'Access to schema:check': [tokenInfo.canCheckSchema ? access.yes : access.not],
-      });
-
-      this.log(print());
-    } else if (result.tokenInfo.__typename === 'TokenNotFoundError') {
-      this.debug(result.tokenInfo.message);
+    if (result.whoAmI == null) {
       throw new InvalidRegistryTokenError();
-    } else {
-      throw new UnexpectedError(
-        `Token response got an unsupported type: ${(result.tokenInfo as any).__typename}`,
-      );
     }
-  }
-}
 
-function createPrinter(records: { [label: string]: [value: string, extra?: string] }) {
-  const labels = Object.keys(records);
-  const values = Object.values(records).map(v => v[0]);
-  const maxLabelsLen = Math.max(...labels.map(v => v.length)) + 4;
-  const maxValuesLen = Math.max(...values.map(v => v.length)) + 4;
+    const data = result.whoAmI;
 
-  return () => {
-    const lines: string[] = [];
+    // Print header
+    this.log(`\n=== ${data.title} ===\n`);
 
-    for (const label in records) {
-      const [value, extra] = records[label];
+    // Iterate and display each permission group
+    for (const permLevel of data.resolvedPermissions) {
+      this.log(`Level: ${permLevel.level}`);
+      this.log(`Resources: ${permLevel.resolvedResourceIds?.join(', ') ?? '<none>'}`);
 
-      lines.push(label.padEnd(maxLabelsLen, ' ') + value.padEnd(maxValuesLen, ' ') + (extra || ''));
-    }
+      const table = new Table({
+        head: ['Group', 'Permission ID', 'Title', 'Granted', 'Description'],
+        wordWrap: true,
+        style: { head: ['cyan'] },
+      });
 
-    return lines.join('\n');
-  };
+      for (const group of permLevel.resolvedPermissionGroups) {
+        for (const perm of group.permissions) {
+          table.push([
+            group.title,
+            perm.permission.id,
+            perm.permission.title,
+            perm.isGranted ? '✓' : '✗',
```

Comment:
It could be nice to do color output also. I find it helps a lot with quickly scanning things.